fedimint-cli/src/lib.rs

```diff
@@ -675,24 +675,16 @@ impl FedimintCli {
         cli: &Opts,
         invite_code: InviteCode,
     ) -> CliResult<ClientHandleArc> {
-        let client_config = cli
-            .connector()
-            .download_from_invite_code(&invite_code)
-            .await
-            .map_err_cli()?;
+        let mut client_builder = self.make_client_builder(cli).await?;
 
-        let client_builder = self.make_client_builder(cli).await?;
+        client_builder.with_connector(cli.connector());
 
         let mnemonic = load_or_generate_mnemonic(client_builder.db_no_decoders()).await?;
 
         let client = client_builder
-            .join(
-                get_default_client_secret(
```

Comment:
Deriving the root secret using federation_id, also something that should be internal to `fedimint-client`.